README.md

```diff
@@ -1,24 +1,163 @@
+> [!NOTE]
+> AAPB-GUID is not [Universally unique identifier](https://en.wikipedia.org/wiki/Universally_unique_identifier), but just a naming convention for AAPB media entries. 
 
-## Source data batches
+### Project Subdirectories 
+Each directory in this repository represents a specific annotation project, its datasets and processing tools.    
+This includes its raw annotated data file, gold-formatted final output data file for tool ingestion, software-suite for converting from raw to gold, and a project-specific `readme.md` explaining it and its annotation guidelines.
 
-A special directory `batches` contains source data batches used in the annotation projects. Specifically, the directory contains `.txt` files named after the batch name. Batch names are usually named after the relevant GitHub issue from [AAPB-CLAMS collaboration repository](https://github.com/clamsproject/aapb-collaboration). Each line in a `.txt` file **must** be a single AAPB GUID, with an exception of any lines starts with `# `, which should be treated as comments.
+The directory name is the name of the project. Each directory contains the following files:
+
+#### Raw intermediate dataset files
+> [!IMPORTANT]
+> `YYMMDD-batchName` directory 
+ 
+These sub-directories contains raw output files from the manual annotation process created by the annotation tool (or by hand like a `.csv` file). 
+Different annotation tools create different file formats with diverse formatting. The `YYMMDD-` prefix _must_ indicate the time when a batch of annotation is conducted. (e.g., when the batch is decided to be annotated)
+These prefixes are important for easy sorting of annotation processes and machine ingestion of the raw data. 
+The `batchName` part of the directory name _must_ match only one of `.txt` files in the batches. 
+
+#### Codebase for conversion
+> [!IMPORTANT]
+> (usually) `process.{sh,py}` and dependencies
+ 
+A piece of software to process the raw manual annotations and generate the publicly-available "gold" dataset. 
+In addition to the main code file, if the code requires additional dependencies/scripts, they can stay in the same directory. 
+The dependencies information can be written down in in the `README.md` (below) file or in a machine-friendly file with the list of dependencies (e.g. `requirements.txt` for `pip`).
+
+#### Final "Gold" dataset files
+> [!IMPORTANT]
+> `golds` directory
+
+This directory contains the public "gold" dataset generated by the above script. The script _must_ generate one file per GUID (video/audio/text document) and the number of gold files in this directory _must_ match the sum of GUIDs in all batches (`YYMMDD-xxx` subdirectories) annotated. 
+The gold dataset is a set of files that are in a format that is ready for use with the newly developed tools. I.e. The raw file must be properly formatted so that tools in the next step of the process can use this dataset.
+
```

Comment:
@owencking raised this question; 
> Can you say what is the expected format for publicly available "gold" datasets? Is it MMIF? Is it some other JSON? Does it vary by tool? Some additional comment would be helpful here, I think.

Regarding it, I suggest update this section to 

---
This directory contains the public "gold" dataset generated by the above script. 

The gold dataset is a set of files that are in a format that is ready for machine consumption primarily for 
1. training ML models for CLAMS apps, 
2. evaluation of CLAMS app outputs, 
3. other public usage

In other words, the distinction between `raw` and `golds` are purely for machine consumption, and the documentation/specification on the organization under `golds` are purely for developers who need to consume those data in their software.

To ensure consistency between data consumption software, there are a few requirements for the `process.py`. 
1. The script _must_ generate one file per GUID (video/audio/text document) 
2. The number of gold files in this directory _must_ match the sum of GUIDs in all batches (`YYMMDD-xxx` subdirectories) annotated. 
    * Namely, there must not be any overlap between assets in batches. 
3. `golds` directory _must not_ have subdirectories. 

That said, `golds` file format must be a common machine-readable data format (CSV, JSON, [definitely not MMIF](https://github.com/clamsproject/mmif/issues/153#issuecomment-1485513488)), and open to change for any future requirements in the consumption software.

Reply:
As a reader of the readme, if I want to get a good idea of what the data looks like, is it better to look at the gold or the raw? 
I feel like its better to look at the gold, especially since its cleaned up. 

(Addednum thoughts: In other projects I've worked on, raw data isnt even publicly seen, just to reduce confusion and explanation. 
In fact, I think we say the same thing: the gold here is meant to be the publicly available info. Yet people may need to see the raw process to understand the workflow.... what are we doing here? )

Reply:
If you're interested in using this data for your purpose, probably you're right to see the `golds`. I also think that we make remarks on the fact that gold data format is subject to change, hence _if one uses our data, they must use the data under perma-links_.

I agree on the uses cases for the "raw" data here as well. There's basically only one use case for the raw data than I can think of: if we ever need to re-use the raw file with the same annotation tool (annotation validation, correction, annotate additional fields/aspects), raw files can be loaded to the same tool.

Reply:
Also, I think all the numbered stuff under `process.py` requirements can be go under the `process.py` section.

Reply:
On second thought, I think the above section is over-explained. Is it neccesary?